src/codegen/cfg.rs

```diff
@@ -336,13 +337,20 @@ pub struct ControlFlowGraph {
     current: usize,
 }
 
+#[derive(Clone)]
+pub enum FunctionNumber {
+    SolidityFunction(usize),
+    YulFunction(usize),
+    None,
+}
```

Comment:
The `FunctionNumber` is a nice idea, although I am not keen on the name. This is a reference to the AST, so how about `ASTFunction`?

Reply:
I do wonder if we need a `DefaultConstructor`, `StorageInitializer`, `Modifier` field. Not sure what use that would be though

Reply:
> The `FunctionNumber` is a nice idea, although I am not keen on the name. This is a reference to the AST, so how about `ASTFunction`?

Currently, there is an `Option<usize>` at the CFG funtion_no, so we don't distinguish between Solidity function and AST Function. As Yul function also have a number, we must differentiate yul function numbers and Solidity function numbers, as they represent indexes to different vectors in NS.

Reply:
Any suggestions for the name?

Reply:
Sounds good.